server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -624,6 +624,7 @@ protected Node(
                 additionalSettingsFilter,
                 settingsUpgraders
             );
+            threadPool.setClusterSettings(settingsModule.getClusterSettings());
```

Comment:
Is there a possibility to put the cluster settings in the ctor of threadpool class?

Reply:
clusterSettings depend on thread pool . So we can't put cluster  settings in the ctor of threadpool class.